src/paho/mqtt/client.py

```diff
@@ -4420,6 +4419,7 @@ def _handle_on_connect_fail(self) -> None:
 
     def _thread_main(self) -> None:
         self.loop_forever(retry_first_connection=True)
+        self._thread = None
```

Comment:
I was thinking to wrap call to loop_forever in _thread_main in a try/finally to be sure `_thread` is set to None even in case of error:

```suggestion
        try:
            self.loop_forever(retry_first_connection=True)
        finally:        
            self._thread = None
```

This only makes change in case loop_forever raise an error.

Reply:
HI Pierre. Thanks for the explanation.

I have made the suggested changes.